src/swish/io.ss

```diff
@@ -237,6 +266,11 @@
   (define osi-port-guardian (make-guardian))
   (define osi-port-table (make-weak-eq-hashtable))
 
+  (define (osi-port-closed? p)
+    (unless (osi-port? p)
+      (bad-arg 'osi-port-closed? p))
+    (not (osi-port-handle p)))
+
   (define (osi-port-count) (hashtable-size osi-port-table))
 
   (define print-osi-ports
```

Comment:
It's not safe to do `(< (osi-port-handle p1) (osi-port-handle p2))` on line 287, is it? Less concerning, but related is line 292.

We disable interrupts only while grabbing the keys in the `osi-port-table` on line 280. While we're sorting the vector someone else could come in and call `close-osi-port` on one of the ports and `<` would be sad about `#f`. Looks like a race that would be difficult to reproduce.

Reply:
Thank you for pointing this out! I fixed it and the other print routines that looked at handles when interrupts are active.